hibernate-query-interpreter/src/main/antlr/org/hibernate/query/parser/internal/hql/antlr/HqlParser.g4

```diff
@@ -360,10 +360,10 @@ parameter
 function
 	: standardFunction
 	| aggregateFunction
-	| jpaNonStandardFunction
-	| nonStandardFunction
 	| jpaCollectionFunction
 	| hqlCollectionFunction
+	| jpaNonStandardFunction
+	| nonStandardFunction
```

Comment:
I've double checked on this. In the original version, e.g. `INDEX( l )` would be matched by `nonStandardFunction`, resulting in a `FunctionExpression` node rather than `CollectionIndexFunction`.